src/main/scala/io/iohk/ethereum/mallet/main/OptionParser.scala

```diff
@@ -0,0 +1,58 @@
+package io.iohk.ethereum.mallet.main
+
+import akka.http.scaladsl.model.Uri
+import io.iohk.ethereum.domain.Address
+import io.iohk.ethereum.mallet.common.{Constants, StringUtil}
+
+/** Command line options and positional args parser */
+object OptionParser {
+
+  private implicit val uriRead: scopt.Read[Uri] =
+    scopt.Read.reads(Uri.apply)
+
+  private implicit val addrRead: scopt.Read[Address] =
+    scopt.Read.reads { str =>
+      val bytes = StringUtil.hexToBytes(str)
+      require(bytes.nonEmpty && bytes.length <= Address.Length, "invalid address length")
+      Address(bytes)
+    }
+
+  def apply(args: Seq[String]): Option[ClOptions] = {
+    val parser = new scopt.OptionParser[ClOptions]("scopt") {
```

Comment:
super minor: the string argument here is a `programName`, guess it should be `Constants.AppName` rather than `"scopt"`

Reply:
Not that minor, good catch